### DISCUSSION TITLE
Manipulating Plot Panel Data

### DISCUSSION THREAD

**Discussion by Jackogip** [foxglove/community#792] [Q&A]
A nice feature in the plotting panel would be to manipulate channels/"series" before they are displayed.

This would be useful for things such as converting between different units for example going from mV to V, degrees F to degrees C, or potentially even running a basic low-pass filter in the interface.

I have seen there is the "User Scripts" panel (I haven't tried it yet) but I guess this would allow the user to manipulate the incoming data. However, it would be much nicer to have an equation field directly inside each plot panel "series" where you could do some basic data manipulation.

**Comment by foxymiles** [comment]
We do support some modifiers on series:
https://foxglove.dev/docs/studio/panels/plot#math-modifiers

`@derivative` would be similar to a low pass filter, for example.

**Reply by Hardliu** [reply]
How do I multiply a data field by a coefficient?

**Reply by Hardliu** [reply]
https://foxglove.dev/docs/studio/panels/plot#math-modifiers
These math are very limited and cannot do much more data processing
